ports/mimxrt/machine_uart.c

```diff
@@ -152,6 +152,21 @@ static void  machine_uart_ensure_active(machine_uart_obj_t *uart) {
     }
 }
 
+void machine_uart_set_baudrate(machine_uart_obj_t *uart, uint32_t baudrate) {
+    #if defined(MIMXRT117x_SERIES)
+    // Use the Lpuart1 clock value, which is set for All UART devices.
+    LPUART_SetBaudRate(uart->lpuart, baudrate, CLOCK_GetRootClockFreq(kCLOCK_Root_Lpuart1));
+    #else
+    // For baud rates < 460800 divide the clock by 10, supporting baud rates down to 50 baud.
+    if (baudrate >= 460800) {
```

Comment:
Maybe create a small inline function that does this clock division setting?  Because this code is repeated below in `machine_uart_init_helper`.